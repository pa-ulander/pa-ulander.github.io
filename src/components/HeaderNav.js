import React from 'react'
import { Link } from 'gatsby'
import useDarkMode from 'use-dark-mode'
import ToggleDarkMode from './ToggleDarkMode'

const HeaderNav = () => {
  const darkMode = useDarkMode(true)

  return (
    <div className="sticky-nav">
      <Link to={`/`} className="sticky-nav-button">
        Home
      </Link>
      <div className="sticky-nav-stack">
        <Link to={`/blog`} className="sticky-nav-button">
          Blog
        </Link>
        <Link to={`/cv`} className="sticky-nav-button">
          Cv
        </Link>
      </div>
      <ToggleDarkMode checked={darkMode.value} onChange={darkMode.toggle} />
    </div>
  )
}

export default HeaderNav
