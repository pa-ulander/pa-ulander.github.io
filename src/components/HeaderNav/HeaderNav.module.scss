.nav {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  font-size: 1.2rem;
  width: 100%;
  height: 75px;
  padding: 1rem;
  position: fixed;
  z-index: 10;
  top: 0;
  background-color: var(--menu-bg-color);
  backdrop-filter: saturate(180%) blur(20px);
  transition: background-color 0.2s ease-in-out;

  ul {
    list-style-type: none;
    margin: 0;
    padding: 0;
    padding-left: 110px;
    justify-content: center;
    padding-right: 110px;
    display: flex;
    flex-grow: 1;
    align-items: flex-end;
  }

  .navButton {
    color: var(--primary-text-color);
    background: transparent;
    border: 1px solid transparent;
    border-radius: 5%;
    padding: 0.5rem;
    margin-right: 1rem;
    text-align: center;
    display: inline-block;
    text-decoration: none;
  }

  .navButton:hover,
  .navButton:active {
    color: #000;
    background: #efefef;
  }
  .toggler {
    padding:0.5rem 0.5rem 0.75rem 0.5rem
  }
}
