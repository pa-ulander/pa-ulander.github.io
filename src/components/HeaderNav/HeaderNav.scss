.nav {
  font-size: 1.2rem;
  width: 100%;
  height: 60px;
  padding: 0.65rem;
  position: sticky;
  top: 0;
  z-index: 10;
  background-color: var(--menu-bg-color);
  backdrop-filter: saturate(180%) blur(20px);
  transition: background-color 0.2s ease-in-out;

  .navprinticon {
    display: none;
  }

  ul {
    display: flex;
    flex-grow: 1;
    align-items: center;
    list-style-type: none;
    justify-content: center;
    margin: 0;
    padding: 0;
    padding-left: 110px;
    padding-right: 110px;
  }

  .navButton {
    color: var(--primary-text-color);
    background: transparent;
    border: 1px solid transparent;
    border-radius: 5%;
    padding: 0.5rem;
    margin-right: 1rem;
    text-align: center;
    display: inline-block;
    text-decoration: none;
  }

  .navButton:hover,
  .navButton__active {
    background: var(--hover-color);
  }

  .toggler {
    padding: 0.5rem;
  }
}

@media (min-width: 576px) {
  .nav {
    .navprinticon {
      display: block;
      position: fixed;
      top: 1rem;
      right: 2rem;
      height: 1.8rem;
      width: 1.8rem;
      color: var(--secondary-text-color);
    }
  }
}

@media print {
  .nav {
    display: none;
  }
}
