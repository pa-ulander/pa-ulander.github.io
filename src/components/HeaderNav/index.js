import React from 'react'
import { Link } from 'gatsby'
import useDarkMode from 'use-dark-mode'
import ToggleDarkMode from '../ToggleDarkMode'
import style from './HeaderNav.module.scss'

const HeaderNav = ({ data }) => {
  const links = [
    { url: `/`, name: `HOME` },
    { url: `/blog`, name: `BLOG` },
    { url: `/cv`, name: `CV` },
  ]

  const darkMode = useDarkMode(true)
  return (
    <div className={style.nav}>
      <ul>
        {links.map((link, i) => (
          <li key={`li-${i}`}>
            <Link key={link.name} to={link.url} className={style.navButton}>
              {link.name}
            </Link>
          </li>
        ))}
        <li className={style.toggler}>
          <ToggleDarkMode checked={darkMode.value} onChange={darkMode.toggle} />
        </li>
      </ul>
    </div>
  )
}

HeaderNav.displayName = 'HeaderNav'

export default HeaderNav
