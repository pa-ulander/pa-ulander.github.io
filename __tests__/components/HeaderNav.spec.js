import React from 'react'
import { shallow } from 'enzyme'
// import renderer from "react-test-renderer"
import HeaderNav from '../../src/components/HeaderNav'

describe('Header component', () => {
  it('should render correctly', () => {
    const component = shallow(<HeaderNav />)
    expect(component).toMatchSnapshot()
  })
})
