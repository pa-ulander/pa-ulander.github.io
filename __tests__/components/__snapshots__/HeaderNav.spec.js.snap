// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render correctly 1`] = `
<div
  className="sticky-nav"
>
  <mockConstructor
    className="sticky-nav-button"
    to="/"
  >
    Home
  </mockConstructor>
  <div
    className="sticky-nav-stack"
  >
    <mockConstructor
      className="sticky-nav-button"
      to="/blog"
    >
      Blog
    </mockConstructor>
    <mockConstructor
      className="sticky-nav-button"
      to="/cv"
    >
      CV
    </mockConstructor>
  </div>
  <ToggleDarkMode
    checked={true}
    onChange={[Function]}
  />
</div>
`;
